tests/Avalonia.Base.UnitTests/AvaloniaObjectTests_Binding.cs

```diff
@@ -882,7 +882,7 @@ public void Untyped_LocalValue_Binding_Logs_Invalid_Value_Type()
                 source.OnNext(1.2);
                 source.OnNext("foo");
 
-                Assert.Equal(5.6, target.GetValue(Class1.QuxProperty));
+                Assert.Equal(1.2, target.GetValue(Class1.QuxProperty));
```

Comment:
I extended your WPF test a bit. The value changes to the default, BUT it retains its original value source:

```C#
Assert.Equal(BaseValueSource.Local, DependencyPropertyHelper.GetValueSource(target, Class1.DoubleProperty).BaseValueSource);
```

If I provide the initial value via a style binding rather than calling `SetBinding` locally on the object, I get `BaseValueSource.Style` instead.

Looks like this PR just became dependent on #9985!

Reply:
It's by no means an implementation detail. It affects the whole outcome of the operation.

When Avalonia/master can't convert the object produced by a binding, it clears the property's local value. This removes any local bindings. But you don't necessarily get the default value, because frames with lower binding priorities can now take effect.

When WPF can't convert an object produced by a binding, it assigns the default value _as_ the current value, with the current priority. This both guarantees that the value is the default, and doesn't remove any bindings.

I think that this branch should start doing what WPF does.

> Oh wait, Avalonia does still have the correct value after these changes

Huh? Now I'm also confused. The current state of the branch shouldn't do anything to the property.

Reply:
Yeah, sorry, I've not been able to free up enough time to review your PRs properly, I will try to find a bit of uninterrupted time to review them properly. Apologies for that.

Reply:
No worries. I started implementing the WPF behaviour, but found that the changes would significantly increase the complexity of the PR. So instead I'll revert all my changes in this area. What matters for this PR is that we can assign null to a nullable.

For future reference, there are three classes that implement the setting of prioritised binding values:

* `BindingEntryBase`
* `LocalValueBindingObserver`
* `LocalValueUntypedBindingObserver`

All of these need to be changed at the same time. Currently I have only changed one of them, whoops!